pkg/volume/util/fs/fs.go

```diff
@@ -56,6 +57,13 @@ func FsInfo(path string) (int64, int64, int64, int64, int64, int64, error) {
 
 // DiskUsage gets disk usage of specified path.
 func DiskUsage(path string) (*resource.Quantity, error) {
+	// First check whether the quota system knows about this directory
+	data, err := quota.GetConsumption(path)
+	if err == nil {
```

Comment:
If the filesystem (type or instance) in question does not support quotas, then it should not be an error if the quota fails, obviously.  But I don't think we want DiskUsage knowing anything about underlying filesystems.  I suppose we could use *resource.Quantity here, where nil means "the filesystem does not support/have enabled quotas" and an error means "something went wrong".

Reply:
My main point is that I would like to either use quota, or not use quota (based on the filesystem and if quota is enabled).  I would rather not have fallback behavior.

Reply:
Right, so if DiskUsage here returns nil it will mean "this filesystem does not support quotas or quotas are not enabled".  If it returns an error it means "something went wrong".

Reply:
You are checking if `err == nil`, and only returning if that is the case.  If we get an error from the `GetConsumption` call, it won't do anything with it, and will call `exec.Command(..., "du", ...)`.  IIUC, this means we are ignoring errors from the `GetConsumption` call.

Am I missing something?  Do we only return errors when the filesystem doesn't support quotas?  What happens if we fail the quotactl syscall?

Reply:
That would work, although i'm not a fan of values with implicit meanings.  Could we just do the `SupportsQuota` check here instead of inside the `GetConsumption` call?

Reply:
We certainly could; I just wanted to keep usage of quota API to as few places as possible.  But this does already make one quota API call, and we're already exposing SupportsQuotas, so it's not really much of a problem.

Reply:
Done per my last comment.